editing/edit-with-branch-versioning/src/main/java/com/esri/samples/edit_with_branch_versioning/EditWithBranchVersioningController.java

```diff
@@ -332,13 +334,18 @@ private void selectFeature(Point2D point) {
    */
   private void updateFeature(ArcGISFeature selectedFeature) {
     if (serviceFeatureTable.canUpdate(selectedFeature) && !serviceGeodatabase.getVersionName().equals(defaultVersion)) {
+
       // update the feature in the feature table
-      ListenableFuture<Void> editResult = serviceFeatureTable.updateFeatureAsync(selectedFeature);
-      editResult.addDoneListener(() -> {
-        Alert alert = new Alert(Alert.AlertType.INFORMATION);
-        alert.setHeaderText("Feature updated");
-        alert.setContentText("Changes will be synced to the service geodatabase\nwhen you switch version.");
-        alert.show();
+      ListenableFuture<Void> updateFuture = serviceFeatureTable.updateFeatureAsync(selectedFeature);
+      updateFuture.addDoneListener(() -> {
+        try {
+          Alert alert = new Alert(Alert.AlertType.INFORMATION);
+          alert.setHeaderText("Feature updated");
+          alert.setContentText("Changes will be synced to the service geodatabase\nwhen you switch version.");
+          alert.show();
+        } catch (Exception ex) {
```

Comment:
I think this all looks alright. Other samples usually put a call to `applyEdits` in the `.addDoneListener`, but since we're just doing the 'apply' when versions are switched, this is ok.

Another thing I've noticed is that `updateFeatureAsync` only throws one type of exceptions:
`IllegalArgumentException - if feature is null`:

https://developers.arcgis.com/java/latest/api-reference/reference/com/esri/arcgisruntime/data/FeatureTable.html#updateFeatureAsync(com.esri.arcgisruntime.data.Feature)

So, perhaps if you include a null check before this method is called (line89?) we can get rid of the try/catch alltogether?

Reply:
Ah nice. I've added into the existing if statement in that selection listener as seemed to make sense there.